src/porepy/models/run_models.py

```diff
@@ -21,71 +34,85 @@ def run_stationary_model(model, params):
     model.after_simulation()
 
 
-def run_time_dependent_model(model, params):
+def run_time_dependent_model(model, params) -> None:
     """
-    Time loop for the model classes.
+    Run a time dependent model.
 
     Args:
         model: Model class containing all information on parameters, variables,
             discretization, geometry. Various methods such as those relating to solving
             the system, see the appropriate solver for documentation.
         params: Parameters related to the solution procedure. # Why not just set these
             as e.g. model.solution_parameters.
+
     """
     # Assign parameters, variables and discretizations. Discretize time-indepedent terms
     if params.get("prepare_simulation", True):
         model.prepare_simulation()
 
-    # Prepare for the time loop
-    t_end = model.end_time
-    model.time_index = 0
+    # Assign a solver
+    solver: Union[pp.LinearSolver, pp.NewtonSolver]
     if model._is_nonlinear_problem():
         solver = pp.NewtonSolver(params)
     else:
         solver = pp.LinearSolver(params)
-    while model.time < t_end:
-        model.time += model.time_step
-        model.time_index += 1
+
+    # Time loop
+    while model.tsc.time < model.tsc.time_final:
+        model.tsc.increase_time()
+        model.tsc.increase_time_index()
         logger.info(
             "\nTime step {} at time {:.1e} of {:.1e} with time step {:.1e}".format(
```

Comment:
Wouldn't we lose the initial time and time index in the logger, by increasing them before storing the info?

Reply:
Isn't the logger deprecated @IvarStefansson @keileg? But to your point @taj004, in this case the logger will not store the initial conditions, only the information starting from `time_index = 1`.  If we want to log the information related to the initial conditions without modifying the current implementation of the time loop, we'll have to add a logger.info() at initialisation.

Reply:
I think it's still in use, but defer to @keileg on this one